src/app/fddev/dev.c

```diff
@@ -61,12 +63,87 @@ install_parent_signals( void ) {
     FD_LOG_ERR(( "sigaction(SIGINT) failed (%i-%s)", errno, fd_io_strerror( errno ) ));
 }
 
+static ushort
+compute_shred_version( char const * genesis_path ) {
+  /* Compute the shred version and the genesis hash */
+  fd_sha256_t _sha[ 1 ];  fd_sha256_t * sha = fd_sha256_join( fd_sha256_new( _sha ) );
+  fd_sha256_init( sha );
+  uchar buffer[ 1024 ];
```

Comment:
```suggestion
  uchar buffer[ 4096 ];
```

Might as well use PAGE_SZ here for perf